packages/internal/src/callback.js

```diff
@@ -155,3 +176,86 @@ export const isCallback = callback => {
   );
 };
 harden(isCallback);
+
+/**
+ * Prepare an attenuator class whose methods can be redirected via callbacks.
```

Comment:
This is great and general purpose, but I wish there was also a way to directly support the degenerate case where we create the attenuator instance from a single target object (which would potentially be marked as sync or async), with no custom method mapping.

I know it's not complicated to map the expected method names of the attenuator to a callback structure, but that requires this method list to be carried around with the attenuator constructor. The attenuator already has the list of method name, so why not let it generate the callbacks if a target object is provided instead?

Reply:
Updated this to use an options bag, basically equivalent to:

```js
const {
  target = null,
  isSync = false,
  overrides = {},
} = opts
```